### DISCUSSION TITLE
Firmware version 6.0 - where to put the version number?

### DISCUSSION THREAD

**Discussion by tingox** [gusmanb/logicanalyzer#152] [Q&A]
I've built the software from the version/v6_0 branch, and also the firmware. In that branch there is no "LogicAnalyzer_Build_Settings.h" file. There is a "LogicAnalyzer_Build_Settings.cmake" file instead, but the contents are (of course) different.

So where do I put the FIRMWARE_VERSION define?

I tried building without, but then the firmware ends up like this
```
$ picotool info LogicAnalyzer.uf2
File LogicAnalyzer.uf2:

Program Information
 name:          LogicAnalyzer
 version:       5.2
 features:      USB stdin / stdout
 binary start:  0x10000000
 binary end:    0x10008258
```
and of course the software complains when I try to connect to the device with this software on
"Cannot connect to device: (Invalid device version LOGIC_ANALYZER_PICO_V5_2, minimum supported version: V6_0)."

**Comment by gusmanb** [comment]
V6_0 is a work in progress and not meant to be used until it's merged into master.

If you want to test the firmware/software use the available [test builds](https://github.com/gusmanb/logicanalyzer/discussions?discussions_q=is%3Aopen+test+build)
